proc/basic_server.cpp

```diff
@@ -41,26 +68,22 @@ int main(void) {
     if(listen(sockfd, 3) < 0) {
         error("ERROR on listen\n");
     }
-    if((newsockfd = accept(sockfd, (struct sockaddr *)&cli_addr,
-                           (socklen_t *)&addrlen)) < 0) {
-        error("ERROR on accept\n");
-    }
 
-    if(read(newsockfd, buf, BUFLEN) < 0) {
-        error("ERROR reading from socket\n");
-    }
+    while((newsockfd = accept(sockfd, (struct sockaddr *)&cli_addr,
+                              (socklen_t *)&addrlen))) {
+        pthread_t sniffer_thread;
+        thsockfd = (int *)malloc(1);
```

Comment:
valgrind produced invalid write size 4 when using this malloc call. weird.
warning goes away when using new opeator.

thsockfd = new int;

Reply:
Fixed.